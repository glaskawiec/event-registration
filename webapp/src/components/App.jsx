import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux'
import Layout from './Layout';
import RegisterEventForm from './RegisterEventForm';
import RegisterEventSuccess from './RegisterEventSuccess';

export const App = (props) => (
    <Layout>
        {
            props.createEventSuccess ?
                <RegisterEventSuccess /> :
                <RegisterEventForm />
        }
    </Layout>
)

App.propTypes = {
    createEventSuccess: PropTypes.bool.isRequired
}

const ConnectedApp = connect(
    state => ({
        createEventSuccess: state.request.createEvent.isSucceeded
    }),
    () => ({})
)(App)

export default ConnectedApp;