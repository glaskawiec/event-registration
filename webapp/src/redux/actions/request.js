import { apiUri } from '../../config/default.json';

import {
    REQUEST_CREATE_EVENT_PENDING,
    REQUEST_CREATE_EVENT_SUCCESS,
    REQUEST_CREATE_EVENT_FAILURE,
    REQUEST_CREATE_EVENT_RESET
} from '../types/request'

export const createEventPending = () => ({
    type: REQUEST_CREATE_EVENT_PENDING
})

export const createEventSuccess = () => ({
    type: REQUEST_CREATE_EVENT_SUCCESS
})

export const createEventFailure = () => ({
    type: REQUEST_CREATE_EVENT_FAILURE
})

export const createEventReset = () => ({
    type: REQUEST_CREATE_EVENT_RESET
})

export const createEvent = (event) => {
    return dispatch => {
        dispatch(createEventPending());
        return fetch(`${apiUri}/events`, {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(event)
        })
            .then((response) => {
                if (response.ok) {
                    dispatch(createEventSuccess())
                } else {
                    dispatch(createEventFailure())
                }
            })
            .catch(() => { dispatch(createEventFailure()) });
    }
}

