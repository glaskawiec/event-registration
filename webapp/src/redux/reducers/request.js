import {
    REQUEST_CREATE_EVENT_PENDING,
    REQUEST_CREATE_EVENT_SUCCESS,
    REQUEST_CREATE_EVENT_FAILURE,
    REQUEST_CREATE_EVENT_RESET
} from '../types/request'

export const initialState = {
    createEvent: {
        isPending: false,
        hadError: false,
        isSucceeded: false,
    }
}

const requestReducer = (state = initialState, action) => {
    switch (action.type) {
        case REQUEST_CREATE_EVENT_PENDING:
            return {
                ...state,
                createEvent: {
                    ...state.createEvent,
                    isPending: true
                }
            }
        case REQUEST_CREATE_EVENT_SUCCESS:
            return {
                ...state,
                createEvent: {
                    ...state.createEvent,
                    isSucceeded: true,
                    isPending: false
                }
            }
        case REQUEST_CREATE_EVENT_FAILURE:
            return {
                ...state,
                createEvent: {
                    ...state.createEvent,
                    isPending: false,
                    hadError: true
                }
            }
        case REQUEST_CREATE_EVENT_RESET:
            return { ...initialState }
        default:
            return state
    }
}

export default requestReducer;