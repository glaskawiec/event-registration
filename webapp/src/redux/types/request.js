export const REQUEST_CREATE_EVENT_PENDING = 'REQUEST_CREATE_EVENT_PENDING'
export const REQUEST_CREATE_EVENT_SUCCESS = 'REQUEST_CREATE_EVENT_SUCCESS'
export const REQUEST_CREATE_EVENT_FAILURE = 'REQUEST_CREATE_EVENT_FAILURE'
export const REQUEST_CREATE_EVENT_RESET = 'REQUEST_CREATE_EVENT_RESET'