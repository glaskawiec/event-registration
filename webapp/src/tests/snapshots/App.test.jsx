import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store' //ES6 modules
import App from '../../components/App';
import { initialState as requestInitialState } from '../../redux/reducers/request';

const middlewares = []
const mockStore = configureStore(middlewares);

test('App snapshot', () => {
    const initialState = {
        request: requestInitialState
    }
    const store = mockStore(initialState)
    const tree = renderer
        .create(
            <Provider store={store}>
                <App />
            </Provider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});