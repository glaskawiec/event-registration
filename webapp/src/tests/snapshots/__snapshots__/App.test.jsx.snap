// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  className="center container-fluid"
>
  <div
    className="d-flex justify-content-center row"
  >
    <div
      className="col-xl-4 col-lg-5 col-md-5 col-sm-7"
    >
      <h1
        style={
          Object {
            "margin": 30,
            "textAlign": "center",
          }
        }
      >
        Event Registration
      </h1>
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            First Name
          </label>
          <input
            className="form-control"
            name="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Last Name
          </label>
          <input
            className="form-control"
            name="lastName"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Email
          </label>
          <input
            className="form-control"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Date
          </label>
          <div
            className="rw-datetime-picker rw-widget"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            tabIndex="-1"
          >
            <div
              className="rw-widget-picker rw-widget-container"
            >
              <input
                aria-activedescendant={null}
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns=""
                autoComplete="off"
                className="rw-widget-input rw-input"
                id="rw_1_input"
                onBlur={[Function]}
                onChange={[Function]}
                role="combobox"
                tabIndex="0"
                type="text"
                value=""
              />
              <span
                className="rw-select rw-select-bordered"
              >
                <button
                  aria-label="Select date"
                  className="rw-btn rw-btn-select"
                  onClick={[Function]}
                  tabIndex="-1"
                  title="Select date"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    className="rw-i rw-i-calendar"
                  />
                </button>
              </span>
            </div>
          </div>
        </div>
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Register
        </button>
      </form>
    </div>
  </div>
</div>
`;
